scripts/uberenv/packages/netcdf-c/package.py

```diff
@@ -0,0 +1,237 @@
+    # https://github.com/Unidata/netcdf-c/issues/250
+    depends_on('hdf5@:1.8.999', when='@:4.4.0')
+
+    # The feature was introduced in version 4.1.2
+    # and was removed in version 4.4.0
+    # conflicts('+cdmremote', when='@:4.1.1,4.4:')
+
+    # The features were introduced in version 4.1.0
+    conflicts('+parallel-netcdf', when='@:4.0')
+    conflicts('+hdf4', when='@:4.0')
+
+    @property
+    def force_autoreconf(self):
+        # The patch for 4.7.0 touches configure.ac.
+        return self.spec.satisfies('@4.7.0')
+
+    def configure_args(self):
+        cflags = []
+        cppflags = []
+        ldflags = []
+        libs = []
+
+        config_args = ['--enable-v2',
+                       '--enable-utilities',
+                       '--enable-static',
+                       '--enable-largefile',
+                       '--enable-netcdf-4']
+
+        # The flag was introduced in version 4.1.0
+        if self.spec.satisfies('@4.1:'):
+            config_args.append('--enable-fsync')
+
+        # The flag was introduced in version 4.3.1
+        if self.spec.satisfies('@4.3.1:'):
+            config_args.append('--enable-dynamic-loading')
+
+        config_args += self.enable_or_disable('shared')
+
+        if '~shared' in self.spec or '+pic' in self.spec:
+            # We don't have shared libraries but we still want it to be
+            # possible to use this library in shared builds
+            cflags.append(self.compiler.cc_pic_flag)
+
+        config_args += self.enable_or_disable('dap')
+        # config_args += self.enable_or_disable('cdmremote')
+
+        # if '+dap' in self.spec or '+cdmremote' in self.spec:
+        if '+dap' in self.spec:
+            # Make sure Netcdf links against Spack's curl, otherwise it may
+            # pick up system's curl, which can give link errors, e.g.:
+            # undefined reference to `SSL_CTX_use_certificate_chain_file
+            curl = self.spec['curl']
+            curl_libs = curl.libs
+            libs.append(curl_libs.link_flags)
+            ldflags.append(curl_libs.search_flags)
+            # TODO: figure out how to get correct flags via headers.cpp_flags
+            cppflags.append('-I' + curl.prefix.include)
+
+        if self.spec.satisfies('@4.4:'):
+            if '+mpi' in self.spec:
+                config_args.append('--enable-parallel4')
+            else:
+                config_args.append('--disable-parallel4')
+
+        if self.spec.satisfies('@4.3.2:'):
+            config_args += self.enable_or_disable('jna')
+
+        # Starting version 4.1.3, --with-hdf5= and other such configure options
+        # are removed. Variables CPPFLAGS, LDFLAGS, and LD_LIBRARY_PATH must be
+        # used instead.
+        hdf5_hl = self.spec['hdf5:hl']
+        cppflags.append(hdf5_hl.headers.cpp_flags)
+        ldflags.append(hdf5_hl.libs.search_flags)
+        if hdf5_hl.satisfies('~shared'):
```

Comment:
I added these two lines (184,185).. I will be creating a spack PR soon.

Reply:
"Replace use of --with-hdf5= and other such configure options that violate conventions and causes build problems. Set environment variables CPPFLAGS, LDFLAGS, and LD_LIBRARY_PATH instead, before running configure script. "

...why

Reply:
I cannot comprehend the logic of whoever did that in netcdf... it's actively antagonistic to their users and goes against all general autoconf practices.